ratis-common/src/main/java/org/apache/ratis/protocol/SetConfigurationRequest.java

```diff
@@ -29,25 +29,41 @@ public class SetConfigurationRequest extends RaftClientRequest {
 
   public enum Mode {
     SET_UNCONDITIONALLY,
-    ADD
+    ADD,
+    CAS
   }
 
   public static final class Arguments {
     private final List<RaftPeer> serversInNewConf;
     private final List<RaftPeer> listenersInNewConf;
+    private final List<RaftPeer> serversInCurConf;
+    private final List<RaftPeer> listenersInCurConf;
     private final Mode mode;
 
-    private Arguments(List<RaftPeer> serversInNewConf, List<RaftPeer> listenersInNewConf,Mode mode) {
+    private Arguments(List<RaftPeer> serversInNewConf, List<RaftPeer> listenersInNewConf, Mode mode) {
+      this(serversInNewConf, listenersInNewConf, Collections.emptyList(), Collections.emptyList(), mode);
+    }
+
+    private Arguments(List<RaftPeer> serversInNewConf, List<RaftPeer> listenersInNewConf,
+        List<RaftPeer> serversInCurConf, List<RaftPeer> listenersInCurConf, Mode mode) {
```

Comment:
Let's move the mode in the middle so that it harder to pass incorrect parameters.

Reply:
done